mixins/segment-mixin.ts

```diff
@@ -0,0 +1,57 @@
+declare global {
+  interface Window {
+    digitalData: any
+    _analytics: any,
+    bluemixAnalytics: any
+  }
+}
+
+export const segmentMixin = {
+  created () {
+    const self = this as any
+
+    if (process.client) {
+      window.digitalData = {
+        page: {
+          pageInfo: {
+            productTitle: self.title,
+            analytics: {
+              category: 'Qiskit.org'
+            }
+          }
+        }
+      }
+
+      window._analytics = {
+        segment_key: 'ffdYLviQze3kzomaINXNk6NwpY9LlXcw',
+        coremetrics: false,
+        optimizely: false,
+        googleAddServices: false,
+        fullStory: false,
+        autoPageEventSpa: false,
+        autoFormEvents: false,
+        autoPageView: false
+      }
+    }
+  },
+  beforeRouteEnter (to, from, next) {
+    next((self) => {
```

Comment:
Since this parameter is the page component, can we call it `pageComponent` or `page` or something more meaningful.